### DISCUSSION TITLE
How to get a single item in storage layer with fs or github adapter

### DISCUSSION THREAD

**Discussion by learntheropes** [nitrojs/nitro#1455] [Q&A] [answered]
I have a reproduction for the storage layer where I want to use `fs` on development and `github` on production, where the base is a folder containing multiple files, like so:
```js
export default defineNitroConfig({
  devStorage: {
    db: {
      driver: 'fs',
      base: './lang',
    },
  },
  storage: {
    db: {
      driver: 'github',
      repo: 'learnetheropes/nitro-storage-adapter',
      branch: 'main',
      dir: '/lang',
    },
  },
});
```
```
lang
  - es.json
  - en.json
```
How do I get/set a specific property in one of the json files in the folder?

I can get the whole file in the folder like so:
```js
const { hallo } = await useStorage('db').getItem(`${lang}.json`);
```
But I can't get a specific property, because the following code get the error: `ENOTDIR: not a directory, open '/home/projects/github-b1epcb/lang/es.json/hallo'`
```js
const hallo = await useStorage('db').getItem(`${lang}.json:hallo`);
```
Also, if I try to get all the keys in the file, I get an empty array, when in reality there is an `hallo` key.
```js
 const keys = await useStorage('db').getKeys(`${lang}.json`);
 console.log(keys); // []
```
Getting the whole file and deconstruct the property is doable, but until I don't solve this, I'm not able to use `setItem` of a property in the file.

Reproduction where I successfully retrieve the whole file:
https://stackblitz.com/edit/github-b1epcb?file=routes%2Ffile%2F[lang].get.js&initialpath=/file/es

Reproduction where I fail to retrieve a specific key/item
https://stackblitz.com/edit/github-b1epcb?file=routes%2Fproperty%2F[lang].get.js&initialpath=/property/es

**Comment by Hebilicious** [comment, marked as answer]
Load the json file first. `getItem` will parse the json file into a JS object, so you can access its property directly.

```ts    
  const json = await useStorage('db').getItem(`${lang}.json`);
  return json.hallo;
```
I don't understand, what are you trying to achieve ?
If you want to modify that property with `setItem`, you have to pass the entire modified file, not just the property : 

```ts    
  const json = await useStorage('db').getItem(`${lang}.json`);
  const newJson = {...json, hallo: "hello there"}
  await useStorage('db').setItem(`${lang}.json`, newJson);
```

**Reply by learntheropes** [reply]
Thank you.

For `fs` and `github` this syntax for `defineNitroConfig` works:
```js
export default defineNitroConfig({
  devStorage: {
    db: {
      driver: 'fs',
      // ...
    },
  },
  storage: {
    db: {
      driver: 'github',
      // ...
    },
  },
}); 
```
From your answer, I understand that is confirmed I can't get or set a single property from the json file.

Instead, I have to get or set the complete file.
